### DISCUSSION TITLE
passing arguments to callbacks

### DISCUSSION THREAD

**Discussion by jkissi** [marcomusy/vedo#604] [Q&A] [answered]
Hi marco, 
I am aware of the callback signature that passes a function name, where that function listens for an event to be processed but is there a way to pass extra arguments to addCallback?

vis = ve.plotter.Plotter(bg='black', size=(1080, 768))
vis.addCallback('mouse move', func)

Also, in the above example, does the addCallback function know about the plotter object? Is that passed automatically? 

Cheers, 
j

**Comment by marcomusy** [comment, marked as answer]
There is no way to pass extra arguments... but I cannot think of a case where this is really necessary, given that you can have variables in same scope of callback or access them in a class structure.
If the variable is in a different file you can still import it..
